File: apps/demo-be-java-springboot/src/main/java/com/organiclever/be/attachment/repository/AttachmentRepository.java
package com.organiclever.be.attachment.repository;

import com.organiclever.be.attachment.model.Attachment;
import com.organiclever.be.expense.model.Expense;
import java.util.List;
import java.util.Optional;
import java.util.UUID;
import org.springframework.data.jpa.repository.JpaRepository;

public interface AttachmentRepository extends JpaRepository<Attachment, UUID> {
    List<Attachment> findAllByExpense(Expense expense);

    Optional<Attachment> findByIdAndExpense(UUID id, Expense expense);
}

File: apps/demo-be-ktkt/src/main/kotlin/com/organiclever/demoktkt/infrastructure/repositories/AttachmentRepository.kt
package com.organiclever.demoktkt.infrastructure.repositories

import com.organiclever.demoktkt.domain.Attachment
import java.util.UUID

data class CreateAttachmentRequest(
  val expenseId: UUID,
  val userId: UUID,
  val filename: String,
  val contentType: String,
  val sizeBytes: Long,
  val storedPath: String,
)

interface AttachmentRepository {
  suspend fun create(request: CreateAttachmentRequest): Attachment

  suspend fun findById(id: UUID): Attachment?

  suspend fun findAllByExpense(expenseId: UUID): List<Attachment>

  suspend fun delete(id: UUID): Boolean
}
